File: twitch/cmd/gpt.py
import asyncio
from utils.split_by_words import split_by_words

async def gpt_cmd(self, ctx):
    if ctx.author.name in self.config['ADMIN'] or ctx.author.name in self.config['PRENIUM']:
        user_input = (
            "donne une réponse courte : "
            + ctx.message.content[len(ctx.prefix) + len(ctx.command.name) + 1 :]
        )

        response = self.client_openia.chat.completions.create(
            model="gpt-4-1106-preview",
            messages=[{"role": "user", "content": user_input}],
            max_tokens=500,
        )

        gpt_text = response.choices[0].message.content.strip()

        chunks = split_by_words(gpt_text)

        for chunk in chunks:
            await ctx.send(f"@{ctx.author.name} {chunk}")
            await asyncio.sleep(1) 

File: src/cmd/gpt.ts
import tmi from 'tmi.js';
import type { Config } from '../types';
import OpenAI from 'openai';

const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY
});

export const gpt_cmd = async (BOT:tmi.Client, channel:string, author:string, config:Config, message:string) => {
    console.log(`CMD : ${author} a utilisé la cmd gpt`);
    if (config.ADMIN.includes(author) || config.PRENIUM.includes(author)) {
        const userInput = message.slice('!gpt'.length).trim();
        const prompt = 'donne une réponse courte : ' + userInput;

        const reponse = await openai.chat.completions.create({
            model: 'gpt-4-1106-preview',
            messages: [{role: 'user', content: prompt}],
            max_tokens: 200
        });

        const gptText = reponse.choices[0]?.message?.content?.trim();
        if (!gptText) return;

        BOT.say(channel, `@${author} ${gptText}`);
    }
};